packages/instantsearch-react/src/InstantSearch.js

```diff
@@ -259,24 +264,35 @@ class InstantSearch extends Component {
     const baseSearchParameters = new SearchParameters({
       index: this.props.indexName,
     });
+    // @TODO: Provide a way to either configure base SearchParameters.
     const searchParameters = this.widgetsManager.getSearchParameters(
       baseSearchParameters
     );
-    this.helper.searchOnce(searchParameters)
-      .then(({content}) => {
-        this.store.setState({
-          ...this.store.getState(),
-          results: content,
-          searching: false,
-        });
-      })
-      .catch(error => {
-        this.store.setState({
-          ...this.store.getState(),
-          error,
-          searching: false,
-        });
+    const promise = this.helper.searchOnce(searchParameters);
+
+    promise.then(({content}) => {
+      this.store.setState({
+        ...this.store.getState(),
+        results: content,
+        searching: false,
+      });
+    }).catch(error => {
+      // Since setState is synchronous, any error that occurs in the render of a
+      // component will be swallowed by this promise.
+      // This is a trick to make the error show up correctly in the console.
+      // See http://stackoverflow.com/a/30741722/969302
+      setTimeout(() => {
```

Comment:
`console.error()` creates two stack traces.

![image](https://cloud.githubusercontent.com/assets/1621758/17592072/2862cf40-5fe1-11e6-9437-6588f31aa132.png)

 This trick allows for the error to appear as if it was thrown outside of a promise.